src/sdk/nsfs_glacier_backend/tapecloud.js

```diff
@@ -133,16 +132,22 @@ class TapeCloudGlacierBackend extends GlacierBackend {
                 // Skip the file if it shouldn't be migrated
                 if (!should_migrate) return true;
 
-                await filtered_log.append(entry);
+                await filtered_log.append(entry.path);
                 return true;
             });
 
             // If the result of the above is false then it indicates that we concluded
             // to exit early hence the file shouldn't be processed further, exit
             if (!result) return false;
 
+            // If we didn't read even one line then it most likely indicates that the WAL is
+            // empty - this case is unlikely given the mechanism of WAL but still needs to be
+            // handled.
+            // Still return `false` so as to not insist file deletion
+            if (processed === 0) return false;
```

Comment:
Ok, so this file will keep being read every time we process migrations?
I think it's best to clean up here.
If you are worried about this case hiding some other bug so lets add a log warn message that we can look out for.
In this case you can also print some more info about the file - like it's stat attributes (size, dates, etc).
But I don't think that leaving it is better than not.